### PR TITLE
Add gcm_sender_id prop on ManifestOptions

### PR DESCRIPTION
**Pull request by ghashi**
Just adding `gcm_sender_id` prop on `ManifestOptions`

related: https://github.com/arthurbergmz/webpack-pwa-manifest/issues/46

### CODE REVIEW

**Comment by frlinw** [comment]
`gcm_sender_id` is legacy stuff and shouldn't by used

**Comment by ghashi** [comment]
@frlinw could you please provide some documentation on that?

I am following this [doc](https://developers.google.com/web/ilt/pwa/introduction-to-push-notifications#sending_a_push_message_using_firebase_cloud_messaging) in order to setup `Push Notifications` on my PWA. And it says:

> For Chrome to route FCM messages to the correct service worker, it needs to know the Sender ID. Supply this by adding a gcm_sender_id property to your app's manifest.json file. 

ps: it also says: 
> Note: The gcm_sender_id is required for Chrome prior to version 52, Opera Android, and Samsung Internet.

Since this config is legacy, but it is still working and documented, I think this could be accepted for now. What do you think? I will add a comment saying it is legacy (adding the documentation you'll recommend on that) in the code...

**Comment by frlinw** [comment]
comment added here for information https://github.com/arthurbergmz/webpack-pwa-manifest/issues/46#issuecomment-370541516

Chrome 52 is a release from 2016/07, focus on last 2 (or 3) versions for evergreen browsers is a best practice.

If you want to start with web push, try this one https://web-push-book.gauntface.com/ (gauntface is a googler working on Workbox)

Anyway, custom properties in pwa-manifest options are added as-is in the output manifest.json

File: index.d.ts [hunk 1]
```diff
@@ -34,6 +35,7 @@ declare namespace WebpackPwaManifest {
         theme_color?: string;
         'theme-color'?: string;
         ios?: boolean | IosOptions;
+        publicPath?: string;
```
Comment (ghashi) [inline index.d.ts]: from the `README.md`: 

> If `publicPath` option is not given, this plugin fallbacks to [Webpack's public path](https://webpack.js.org/configuration/output/#output-publicpath) definition.

**Comment by ghashi** [comment]
@frlinw thank you very much for the references! =)
I'm still reading https://web-push-book.gauntface.com/ .
It is very well written.

Well, so it doesn't make sense to add `gcm_sender_id`. Therefore I'll close this PR and open another one with the `publicPath` prop. 

Is this ok for you?

**Comment by frlinw** [comment]
@ghashi do not hesitate to bookmark the service worker cookbook from mozilla too https://serviceworke.rs/
And a lil useful tool to generate VAPID easily https://web-push-codelab.glitch.me/

About the PR, I let @arthurbergmz answer

**Comment by arthurbergmz** [comment]
Not adding `gcm_sender_id`, but I updated our typing with `publicPath`. Thanks!

**Pull request closed without merging**